ranger/ext/rifle.py

```diff
@@ -389,45 +394,39 @@ def execute(self, files,  # noqa: E501 pylint: disable=too-many-branches,too-man
                                          "Please set $TERMCMD manually or "
                                          "change fallbacks in rifle.conf.")
                         self._mimetype = 'ranger/x-terminal-emulator'
-                        self.execute(
-                            files=[command.split(';')[1].split('--')[0].strip()]
-                            + files, flags='f',
-                            mimetype='ranger/x-terminal-emulator')
+                        self.execute(files=cmd, flags='f',
+                                     mimetype='ranger/x-terminal-emulator')
                         return None
 
                     # Choose correct cmdflag accordingly
                     if term in ['xfce4-terminal', 'mate-terminal',
-                                'terminator']:
-                        cmdflag = '-x'
+                                'terminator', 'sakura']:
+                        termflags = ['-x']
                     elif term in ['xterm', 'urxvt', 'rxvt', 'lxterminal',
                                   'konsole', 'lilyterm', 'cool-retro-term',
                                   'terminology', 'pantheon-terminal', 'termite',
                                   'st', 'stterm']:
-                        cmdflag = '-e'
+                        termflags = ['-e']
                     elif term in ['gnome-terminal', 'kitty']:
-                        cmdflag = '--'
-                    elif term in ['tilda', ]:
-                        cmdflag = '-c'
+                        termflags = ['--']
+                    elif term in ['tilda']:
+                        termflags = ['-c']
+                    elif term in ['guake']:
+                        # put pwd manually here? this won't get expanded
```

Comment:
Why does it not get expanded? Might be a reason to stick with `set`?

Reply:
These arguments are passed to `Popen` and not to a shell, so they can't get expanded. I didn't change this part, and wouldn't want to wrap the whole command  in `sh -c ...` again. Just putting `os.environ['PATH']` there should work.

The `-n` flag in `guake` was fixed only recently, so no matter the argument, the `PWD` of the command was always set to `HOME`.

Reply:
Do you see an easy way to use the directory of the files rifle was triggered on instead of the `PWD`? Ranger doesn't change the `PWD` while navigating afair.

Reply:
The fist thing I tried was taking the directory of the first file. While that worked, I realized that this does not match the behaviour of the other terminals. They all inherit `PWD` from ranger, only `guake` needs it for the `-n` flag or it will start in `HOME`.

Reply:
Hmm, maybe we should do `cd /path/to/selected/files; command-found-in-rifle.conf`?
I'm thinking this would mostly be nice with editors, like vim. You'd open a file in a repo, with flag t, then expect to be able to do `:e other-file`, rather than `:e /path/to/repo/other-file`.

Reply:
This would make sense when rifle is started from within ranger, but it's going to break things when used as a standalone script. The Files don't have to be in the same directory anyway and more importantly, relative paths miss after cd'ing.

Reply:
Maybe we should make the paths absolute. I still think it'd be nice to rifle some files and have vim's working directory be set to that of either the first or last file passed, I prefer first but whatever as long as it's consistent.

Do you disagree the vim example is compelling?

Reply:
> Maybe we should make the paths absolute.

I didn't read the whole discussion but if you want to make all the paths passed to `rifle` absolute, it's a Really Bad Idea™. For some utilities, like `tar(1)`, they way the path gets passed is crucial.